### ISSUE TITLE
Plugins cannot specify the entry option

### ISSUE THREAD

**Issue by Pauan** [rollup/rollup#1270]
Here is an example Rollup configuration:

```js
export default {
  plugins: [
    {
      options: function (rollup) {
        rollup.entry = "foo.js";
      }
    }
  ]
};
```

When using `rollup --config` I get the following error:

```
🚨   You must specify an --input (-i) option
```

If I instead use this config, then it works:

```js
export default {
  entry: "foo.js",
  plugins: [
    {
      options: function (rollup) {
        rollup.entry = "foo.js";
      }
    }
  ]
};
```

Basically, it seems that Rollup is checking whether the `entry` option is specified or not, but it does this check before the plugins are run.

I think Rollup should do the check after the plugins run, because a plugin might set the `entry` in an `options` hook.

**Comment by Rich-Harris** [comment]
This is fixed in 0.43, thanks

**Comment by pengliheng** [comment]
........   i also have this problom   
but this issue cant make me know how to fix this problem
